projects/ngneat/reactive-forms/src/lib/formArray.ts

```diff
@@ -190,4 +190,18 @@ export class FormArray<T = any, E extends object = any> extends NgFormArray {
   setDisable(disable = true, opts?: ControlEventOptions) {
     disableControl(this, disable, opts);
   }
+
+  remove(value: T) {
+    this.removeWhen(v => v.value === value);
+  }
+
+  removeWhen(predicate: (element: AbstractControl<T>) => boolean) {
```

Comment:
- I would call it `removeIf`. 
- Why do you need the sorting functionality? Loop over the controls and remove them based on the predicate.

Reply:
The Angular `FormArray.prototype.removeAt` method is in-place.
So the array length will change during the loop when removing elements.
That's why it is required to remove elements from the higher index to the lower.

However good point, it's more straightforward and takes less time to iterate from the last to the first index and directly remove controls.